haskell/experimental/private/module.bzl

```diff
@@ -532,20 +564,75 @@ def _merge_narrowed_deps_dicts(rule_label, narrowed_deps):
                 str(rule_label),
                 str(dep.label),
             ))
+        abis.update(lib_info.per_module_abi)
         _merge_depset_dicts(transitive_interfaces, lib_info.per_module_transitive_interfaces)
         _merge_depset_dicts(transitive_objects, lib_info.per_module_transitive_objects)
         _merge_depset_dicts(transitive_dyn_objects, lib_info.per_module_transitive_dyn_objects)
     return struct(
+        abis = abis,
         transitive_interfaces = transitive_interfaces,
         transitive_objects = transitive_objects,
         transitive_dyn_objects = transitive_dyn_objects,
     )
 
 def interfaces_as_list(with_shared, o):
     if with_shared:
-        return [o.hi, o.dyn_hi]
+        return [o.hi, o.dyn_hi, o.abi]
     else:
-        return [o.hi]
+        return [o.hi, o.abi]
```

Comment:
Is this change necessary? I would think we shouldn't mix interface and abi files here.